### PR TITLE
Issue 5501: Add new command to parse RG stream file

### LINKED ISSUE #5501: CLI improvement: Add command to parse ReaderGroup stream file
**Problem description**
Add a command to:
1. Parse ReaderGroup stream file to a human readable file for debugging
2. Modify Reader Group stream file: truncate, append, etc. - essentially repair it if corrupted.

**Problem location**
CLI

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by Tristan1900**
**Change log description**  
Adds a new command to CLI to parse the ReaderGroup stream file.

How to use
`./pravega_cli debug parse-rg-stream-file /dir/to/stream/file /dir/to/parsed/file`
**Purpose of the change**  
Fix #5501

**What the code does**  
Read stream file header first (4 bytes type + 4 bytes length) and then read the content based on length. Use Java client deserializer to deserialize the content and output to a file.

**How to verify it**  
Manually tested

### CODE REVIEW

**Review by shrids — APPROVED** [review]
LGTM...

**Comment by Tristan1900** [comment]
Hey @RaulGracia , could you take a look at this PR and see if it looks good to you? I have tested manually and used it couple of times for debugging so it should work. 
Thanks!

**Comment by RaulGracia** [comment]
@Tristan1900 looks like there is an issue with code coverage, once you make it happy we can resume with this PR.

File: cli/admin/src/test/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.Test;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class ReaderGroupStreamFileParsingCommandTest {
+
+    @Test
+    public void testPasswordFileCreatorCommand() throws Exception {
```
Comment (RaulGracia) [inline cli/admin/src/test/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommandTest.java:27]: Looks like a bad name for a test.

File: cli/admin/src/test/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        final String inputFileName = "input";
+        final String outputFileName = "output";
+        byte[] bytes = {0, 0, 0, 0, 0, 0, 0, 0};
```
Comment (RaulGracia) [inline cli/admin/src/test/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommandTest.java:30]: Is there a way to test with a more realistic content that actually has some info about a reader group?

File: cli/admin/src/main/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommand.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * The command takes input stream file and output stream file as argument.
+ * It parses the serialized ReaderGroup Stream file to a human readable file.
```
Comment (RaulGracia) [inline cli/admin/src/main/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommand.java:30]: Please, rephrase this so it is clearer what this command does and why it is useful for. Perhaps mentioning an example or scenario where this command is expected to be used could help.

File: cli/admin/src/main/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommand.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * The command takes input stream file and output stream file as argument.
```
Comment (RaulGracia) [inline cli/admin/src/main/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommand.java:29]: How is the user supposed to the this "input file"?

File: cli/admin/src/main/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommand.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    public ReaderGroupStreamFileParsingCommand(CommandArgs args) {
+        super(args);
+    }
+    
+    @Override
+    public void execute() {
+        try {
+            ensureArgCount(2);
+            String inputFileName = getInputStreamFilename(getCommandArgs().getArgs());
+            String outputFileName = getOutputStreamFilename(getCommandArgs().getArgs());
```
Comment (andreipaduroiu) [inline cli/admin/src/main/java/io/pravega/cli/admin/debug/ReaderGroupStreamFileParsingCommand.java:46]: This should be rewritten to make use of either RevisionedStreamClient/ByteStreamClient or the Admin Gateway (raw read of the segment file). 

How is this file supposed to get to your local filesystem in the first place? How do you download it from Pravega?

**Review by andreipaduroiu — CHANGES_REQUESTED** [review]

**Comment by andreipaduroiu** [comment]
@Tristan1900 What is the status of this PR? Are we still pursuing it?

**Pull request closed without merging**